### PR TITLE
Machine learning testing pipeline

### PR DESCRIPTION
**Pull request by SeanOverton**
The testing pipeline for the Machine Learning training capability that has been introduced.

To summarise; this stage allows for the actual use of the previously generated model from the Batch Process which means values can be predicted in the clinical data of a patient.

### CODE REVIEW

File: src/Model/MachineLearningTester.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+        return True
+
+    def save_into_csv(self, path_to_save):
+        """
+        Saves predicted values into csv.
+        Parameters:
+            str : path_to_save : location to save csv
+        """
+        # TODO: is all this necessary?
```
Comment (pedrooste) [inline src/Model/MachineLearningTester.py:104]: TODO

File: src/View/mainpage/MLTab.py [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
+
+        # once run will trigger results popup window
+        ml_results_window = MLResultsWindow()
+        ml_results_window.set_results_values(results)
+        ml_results_window.set_ml_tester(ml_tester)
+
+        ml_results_window.exec_()
+
+
+class MLResultsWindow(QtWidgets.QDialog):
```
Comment (pedrooste) [inline src/View/mainpage/MLTab.py:517]: good to have a separate file per class

**Review by pedrooste — COMMENTED** [review]
Small comments but overall the code looks good.

Not sure how to test from my end

File: src/View/mainpage/MLTab.py (hunk 2 above)
Comment (SeanOverton) [reply src/View/mainpage/MLTab.py]: True

**Pull request merged**